app/src/main/java/com/example/dynamicmessenger/authorization/viewModels/EmailAndPhoneViewModel.kt

```diff
@@ -21,9 +22,9 @@ class EmailAndPhoneViewModel(application: Application) : AndroidViewModel(applic
             try {
                 val response = MailExistApi.retrofitService.isMailExistAsync(task)
                 if (response.isSuccessful) {
-                    SharedPreferencesManager.setUserMailExists(context!!, response.body()!!.mailExist)
-                    SharedPreferencesManager.setUserCode(context, response.body()!!.code)
-                    SharedPreferencesManager.setUserMail(context, task.email)
+                    MainActivity.userMailExists = response.body()!!.mailExist
+                    MainActivity.userCode = response.body()!!.code
+                    MainActivity.userMail = task.email
```

Comment:
And again these fields should not be in MainActivity. Actually it is not clear their existence meaning.